File: scripts/diagnose_blutorangen.ts
import { createClient } from "@supabase/supabase-js"

const supabaseUrl = process.env.SUPABASE_URL!
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY!
const supabase = createClient(supabaseUrl, supabaseKey)

async function diagnoseBlutorangen() {
  console.log("=== BLUTORANGEN DIAGNOSE ===\n")

  // 1. Find Blutorangen in products table
  const { data: products, error: productsError } = await supabase
    .from("products")
    .select("id, name, is_active, stock_quantity, inventory_raw_id, unit, price, category_id")
    .ilike("name", "%blutorange%")

  if (productsError) {
    console.error("Error fetching products:", productsError)
    return
  }

  console.log('1. PRODUKTE MIT "BLUTORANGE" IM NAMEN:')
  console.log(JSON.stringify(products, null, 2))
  console.log("\n")

  if (!products || products.length === 0) {
    console.log("❌ Keine Blutorangen gefunden!")
    return
  }

  const blutorange = products[0]
  console.log(`2. BLUTORANGEN DETAILS:`)
  console.log(`   - ID: ${blutorange.id}`)
  console.log(`   - Name: ${blutorange.name}`)
  console.log(`   - Aktiv: ${blutorange.is_active}`)
  console.log(`   - Stock Quantity: ${blutorange.stock_quantity}`)
  console.log(`   - Inventory Raw ID: ${blutorange.inventory_raw_id}`)
  console.log(`   - Unit: ${blutorange.unit}`)
  console.log("\n")

  // 2. Check inventory_raw_stock if it has inventory_raw_id
  if (blutorange.inventory_raw_id) {
    const { data: rawStock, error: rawStockError } = await supabase
      .from("inventory_raw_stock")
      .select("*")
      .eq("id", blutorange.inventory_raw_id)
      .single()

    if (rawStockError) {
      console.error("Error fetching raw stock:", rawStockError)
    } else {
      console.log("3. GRAMM-BASIERTES LAGER:")
      console.log(`   - Raw Stock ID: ${rawStock.id}`)
      console.log(`   - Product Group: ${rawStock.product_group}`)
      console.log(`   - Stock Grams: ${rawStock.stock_grams}`)
      console.log(`   - Min Stock Grams: ${rawStock.min_stock_grams}`)
      console.log(`   - Unit Type: ${rawStock.unit_type}`)
      console.log("\n")
    }
  } else {
    console.log("3. LAGER-TYP: Stückbasiert (kein inventory_raw_id)\n")
  }

  // 3. Check product_availability view
  const { data: availability, error: availabilityError } = await supabase
    .from("product_availability")
    .select("*")
    .eq("product_id", blutorange.id)
    .single()

  if (availabilityError) {
    console.error("Error fetching availability:", availabilityError)
  } else {
    console.log("4. PRODUCT_AVAILABILITY VIEW:")
    console.log(JSON.stringify(availability, null, 2))
    console.log("\n")
  }

  // 4. Check category
  if (blutorange.category_id) {
    const { data: category, error: categoryError } = await supabase
      .from("categories")
      .select("id, name, slug, is_active")
      .eq("id", blutorange.category_id)
      .single()

    if (!categoryError && category) {
      console.log("5. KATEGORIE:")
      console.log(`   - Name: ${category.name}`)
      console.log(`   - Slug: ${category.slug}`)
      console.log(`   - Aktiv: ${category.is_active}`)
      console.log("\n")
    }
  }

  console.log("=== DIAGNOSE ABGESCHLOSSEN ===")
}

diagnoseBlutorangen()

File: scripts/diagnose_blutorangen.py
import os
from supabase import create_client, Client

# Supabase connection
supabase_url = os.environ.get("SUPABASE_URL")
supabase_key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY")
supabase: Client = create_client(supabase_url, supabase_key)

print("=== BLUTORANGEN DIAGNOSE ===\n")

# 1. Find Blutorangen product
print("1. Suche nach Blutorangen in products Tabelle:")
products = supabase.table("products").select("*").ilike("name", "%blutorange%").execute()
if products.data:
    for product in products.data:
        print(f"  Produkt gefunden: {product['name']} (ID: {product['id']})")
        print(f"    - is_active: {product.get('is_active')}")
        print(f"    - stock_quantity: {product.get('stock_quantity')}")
        print(f"    - inventory_raw_id: {product.get('inventory_raw_id')}")
        print(f"    - category_id: {product.get('category_id')}")
        print(f"    - min_stock: {product.get('min_stock')}")
        
        # 2. Check inventory_raw_stock if assigned
        if product.get('inventory_raw_id'):
            print(f"\n2. Prüfe gramm-basiertes Lager (ID: {product['inventory_raw_id']}):")
            raw_stock = supabase.table("inventory_raw_stock").select("*").eq("id", product['inventory_raw_id']).execute()
            if raw_stock.data:
                stock = raw_stock.data[0]
                print(f"    - product_name: {stock.get('product_name')}")
                print(f"    - stock_grams: {stock.get('stock_grams')}")
                print(f"    - min_stock_grams: {stock.get('min_stock_grams')}")
            else:
                print("    FEHLER: Kein Lager-Eintrag gefunden!")
        else:
            print("\n2. Kein gramm-basiertes Lager zugeordnet (inventory_raw_id ist NULL)")
        
        # 3. Check product_availability view
        print(f"\n3. Prüfe product_availability View:")
        availability = supabase.table("product_availability").select("*").eq("product_id", product['id']).execute()
        if availability.data:
            avail = availability.data[0]
            print(f"    - is_available: {avail.get('is_available')}")
            print(f"    - status: {avail.get('status')}")
            print(f"    - current_stock: {avail.get('current_stock')}")
        else:
            print("    FEHLER: Produkt nicht in product_availability View!")
        
        # 4. Check category
        if product.get('category_id'):
            print(f"\n4. Prüfe Kategorie (ID: {product['category_id']}):")
            category = supabase.table("categories").select("*").eq("id", product['category_id']).execute()
            if category.data:
                print(f"    - Name: {category.data[0].get('name')}")
        
        print("\n" + "="*50 + "\n")
else:
    print("  FEHLER: Keine Blutorangen gefunden!")

print("\n=== DIAGNOSE ABGESCHLOSSEN ===")
